### PR TITLE
Backport merge fixes added to ENT

### PR DESCRIPTION
**Pull request by tudor-malene**
This is a backport of changes made when merging OS to ENT and fixing integration tests.

### CODE REVIEW

File: node/src/integration-test/kotlin/net/corda/node/services/statemachine/LargeTransactionsTest.kt [hunk 1]
```diff
@@ -97,4 +60,40 @@ class LargeTransactionsTest {
             }
         }
     }
-}
\ No newline at end of file
+}
+
+@StartableByRPC
+@InitiatingFlow
+class SendLargeTransactionFlow(private val hash1: SecureHash,
```
Comment (shamsasari) [inline node/src/integration-test/kotlin/net/corda/node/services/statemachine/LargeTransactionsTest.kt]: You will need revert this change. `enclosedCordapp` requires the flows to be inside the test class

File: testing/node-driver/src/main/kotlin/net/corda/testing/driver/Driver.kt [hunk 2]
```diff
@@ -242,7 +242,7 @@ data class DriverParameters(
         val notarySpecs: List<NotarySpec> = listOf(NotarySpec(DUMMY_NOTARY_NAME)),
         @Deprecated("extraCordappPackagesToScan does not preserve the original CorDapp's versioning and metadata, which may lead to " +
                 "misleading results in tests. Use cordappsForAllNodes instead.")
-        val extraCordappPackagesToScan: List<String> = emptyList(),
+        val extraCordappPackagesToScan: List<String> = listOf("migration"),
```
Comment (shamsasari) [inline testing/node-driver/src/main/kotlin/net/corda/testing/driver/Driver.kt]: Can you create a ticket and add a TODO for finding a different way for getting the migration scripts in. For ENT I guess it was OK to modify the driver like this, but the OS driver is generally the "public" one and I'm not sure we want to modify this way. (Also make sure this change does not land into `release/4` branch in anyway).

File: tools/shell/src/integration-test/kotlin/net/corda/tools/shell/HashLookupCommandTest.kt [hunk 3]
```diff
@@ -81,17 +81,17 @@ class HashLookupCommandTest {
...
-        @Suspendable
-        override fun call(): SecureHash {
-            val me = serviceHub.myInfo.legalIdentities.first().ref(0)
-            val fakeNotary = me.party
-            val builder = DummyContract.generateInitial(1, fakeNotary as Party, me)
-            val stx = serviceHub.signInitialTransaction(builder)
-            serviceHub.recordTransactions(stx)
-            return stx.id
-        }
+@StartableByRPC
```
Comment (shamsasari) [inline tools/shell/src/integration-test/kotlin/net/corda/tools/shell/HashLookupCommandTest.kt]: likewise I think you will need to revert this as well

File: tools/shell/src/integration-test/kotlin/net/corda/tools/shell/InteractiveShellIntegrationTest.kt [hunk 4]
```diff
@@ -399,4 +372,31 @@ class InteractiveShellIntegrationTest {
         }
         assertThat(successful).isTrue()
     }
+}
+
+@Suppress("UNUSED")
+@StartableByRPC
+class NoOpFlow : FlowLogic<Unit>() {
```
Comment (shamsasari) [inline tools/shell/src/integration-test/kotlin/net/corda/tools/shell/InteractiveShellIntegrationTest.kt]: ditto

File: testing/node-driver/src/main/kotlin/net/corda/testing/driver/Driver.kt (hunk 2 above)
Comment (tudor-malene) [reply testing/node-driver/src/main/kotlin/net/corda/testing/driver/Driver.kt]: I will remove it from here and everything will work just as before.

File: tools/shell/src/integration-test/kotlin/net/corda/tools/shell/HashLookupCommandTest.kt (hunk 3 above)
Comment (tudor-malene) [reply tools/shell/src/integration-test/kotlin/net/corda/tools/shell/HashLookupCommandTest.kt]: This is like this on ENT and it works fine.

File: tools/shell/src/integration-test/kotlin/net/corda/tools/shell/InteractiveShellIntegrationTest.kt (hunk 4 above)
Comment (tudor-malene) [reply tools/shell/src/integration-test/kotlin/net/corda/tools/shell/InteractiveShellIntegrationTest.kt]: same

**Comment by tudor-malene** [comment]
> Missing:
> 
> * https://github.com/corda/enterprise/pull/1726/files#diff-650423b3e729b43596717b8b344f7e42R695
> * https://github.com/corda/enterprise/pull/1726/files#diff-4026c5f6082d7202e7be325fb23e60ddR27
> 
> Shouldn't the cash `migrationResource` also be ported? e.g. https://github.com/corda/enterprise/pull/1726/files#diff-0942ea4dbcabca7205a6575fb61bfac1R24

Left this stuff out on purpose as it's not supported by OS.

File: node/src/integration-test/kotlin/net/corda/node/flows/AsymmetricCorDappsTests.kt [hunk 5]
```diff
@@ -85,4 +85,30 @@ class AsymmetricCorDappsTests {
             nodeA.rpc.startFlow(::Ping, nodeB.nodeInfo.singleIdentity(), 1).returnValue.getOrThrow()
         }
     }
+}
+
+@StartableByRPC
+@InitiatingFlow
+class Ping(private val pongParty: Party, val times: Int) : FlowLogic<Unit>() {
```
Comment (shamsasari) [inline node/src/integration-test/kotlin/net/corda/node/flows/AsymmetricCorDappsTests.kt]: You need to remove the flows in the test class

**Comment by tudor-malene** [comment]
Back to you @shamsasari
